### ISSUE TITLE
OpenBSD support

### ISSUE THREAD

**Issue by frankie-loves-jesus** [Medium/phantomjs#253]
Expanding on https://github.com/ariya/phantomjs/issues/10996 -- `npm install phantomjs` on OpenBSD will attempt to fetch the MacOS X binaries when OpenBSD binaries are available (http://openports.se/www/phantomjs):

```
# npm install phantomjs -g
-
> phantomjs@1.9.12 install /usr/local/lib/node_modules/phantomjs
> node install.js

Looks like an `npm install -g`; unable to check for already installed version.
Downloading https://bitbucket.org/ariya/phantomjs/downloads/phantomjs-1.9.8-macosx.zip
Saving to /usr/local/lib/node_modules/phantomjs/phantomjs/phantomjs-1.9.8-macosx.zip
Receiving...
  [====------------------------------------] 9% 0.0s^C#
```

**Comment by nanaya** [comment]
> What problem does this solve? I was under the impression that the mac osx binaries work fine on openbsd? is that not true?

This can't be serious.

FWIW, it's currently getting OSX binary for FreeBSD as well. And no, FreeBSD is not OSX either.

```
$ uname -a
FreeBSD nonoka.myconan.net 10.2-RELEASE FreeBSD 10.2-RELEASE #0 r286666: Wed Aug 12 15:26:37 UTC 2015     root@releng1.nyi.freebsd.org:/usr/obj/usr/src/sys/GENERIC  amd64
$ grep version package.json
  "version": "1.9.18",
$ file ./lib/phantom/bin/phantomjs
./lib/phantom/bin/phantomjs: Mach-O i386 executable
$ ./lib/phantom/bin/phantomjs
-su: ./lib/phantom/bin/phantomjs: cannot execute binary file: Exec format error
```

**Comment by adamdilibero** [comment]
> phantomjs@1.9.19 install /usr/home/apache/hosts/WSBprojects/SX-Ecommerce/node_modules/svg2png/node_modules/phantomjs
> node install.js

PhantomJS not found on PATH
Download already available at /tmp/phantomjs/phantomjs-1.9.8-macosx.zip
Verified checksum of previously downloaded file
Extracting zip contents
Removing /usr/home/apache/hosts/WSBprojects/SX-Ecommerce/node_modules/svg2png/node_modules/phantomjs/lib/phantom
Copying extracted folder /tmp/phantomjs/phantomjs-1.9.8-macosx.zip-extract-1450180123360/phantomjs-1.9.8-macosx -> /usr/home/apache/hosts/WSBprojects/SX-Ecommerce/node_modules/svg2png/node_modules/phantomjs/lib/phantom
Writing location.js file
Done. Phantomjs binary available at /usr/home/apache/hosts/WSBprojects/SX-Ecommerce/node_modules/svg2png/node_modules/phantomjs/lib/phantom/bin/phantomjs

Phantom.js is still getting the Mac OS X version from npm on FreeBSD this isn't right?

**Comment by nicks** [comment]
yes, the last comment said that we are still waiting on an OpenBSD port. If there is a port, we would welcome a pull request that uses it.

**Comment by nicks** [comment]
(the owners of this package don't have a freebsd machine, so we can't test any change ourselves. we would need someone to put together a pr and verify it)

**Comment by WuShell** [comment]
? There has been an OpenBSD port for quite some time:

http://cvsweb.openbsd.org/cgi-bin/cvsweb/ports/www/phantomjs/

I didn't have the time to port phantomjs 2 yet though :/

**Comment by WuShell** [comment]
And about freebsd... you should install the package or from ports, not using npm I guess.

**Comment by jirib** [comment]
Please at least change https://github.com/Medium/phantomjs/blob/master/lib/util.js#L106

This 'or' is really ridiculous. Yes, there are differences between operating system and as "you" do not ship prebuilt packages for all operating systems just fail here. (This comment shows how wrong are presupposed ideas https://github.com/Medium/phantomjs/issues/253#issuecomment-61125716.) So just, fail if your project doesn't have prebuilt package for a platform/OS. Thank you.

**Comment by saydulk** [comment]
![screenshot from 2018-07-05 19 06 50](https://user-images.githubusercontent.com/1058663/42324948-94b81e9e-8086-11e8-84bb-53f50e9925ea.png)
same here
